cmake/Doxygen.cmake

```diff
@@ -60,7 +60,7 @@ if(DOXYGEN_FOUND)
         COMMAND ${CMAKE_COMMAND} -E touch ${DOXYGEN_STAMP_FILE}
         WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
         COMMENT "Generating API documentation with Doxygen" VERBATIM)
-    add_custom_target(doc DEPENDS ${DOXYGEN_STAMP_FILE})
+    add_custom_target(oneDNNdoc DEPENDS ${DOXYGEN_STAMP_FILE})
```

Comment:
Sorry, I should've been more clear in the issue text. The acceptable solution is to have a cmake option. Changing the target name will break lots of things (our automation, packagers' scripts, etc).

Reply:
@rsdubtso, what do you think if we enable this target only if oneDNN is not a sub-project of another project?

Reply:
I would be totally fine with that (we could use the same logic for tests, etc). I just could not find a reliable way to do this (but this does not mean that there is no way to do this).